packages/legend-query/src/stores/QueryBuilderMilestoningHelper.ts

```diff
@@ -0,0 +1,641 @@
+/**
```

Comment:
Honestly, I haven't had time to review fully the logic here yet, I'll come back later.